src/Microsoft.OData.Core/JsonLight/ReorderingJsonReader.cs

```diff
@@ -82,6 +83,85 @@ public override bool CanStream()
             return (this.Value is string || this.Value == null);
         }
 
+
+        /// <summary>
+        /// Asynchronously creates a stream for reading a stream value.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains a <see cref="Stream"/> used to read a stream value.
+        /// </returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1825:Avoid zero-length array allocations.", Justification = "<Pending>")]
+        public override async Task<Stream> CreateReadStreamAsync()
+        {
+            Stream result;
+            object value = null;
+
+            try
+            {
+                value = await this.GetValueAsync()
+                    .ConfigureAwait(false);
+
+                if (value == null)
+                {

```

Comment:
suggestion: Maybe use the Stream.Null to avoid any allocations if we don't reuse this. 
https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.null?view=net-6.0

Reply:
Ps you may need to evaluate if this would be potentially breaking.

Reply:
I'm inclined towards not making use of `Stream.Null` at this time to allow myself to understand what effect and impact that might have. I'd also not want the logic in async method to differ too much from the sync method. I'll see about doing a PR that addresses from the sync and async implementations.